common/src/gas_provider/node.rs

```diff
@@ -19,6 +19,29 @@
 use super::*;
 use codec::MaxEncodedLen;
 
+/// ID of the [`GasNode`].
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Encode, Decode, TypeInfo)]
+pub enum GasNodeId<T, U> {
+    Node(T),
+    Reservation(U),
+}
+
+impl<T, U> GasNodeId<T, U> {
+    pub fn to_node_id(self) -> Option<T> {
```

Comment:
> Prefix | Cost | Ownership
-- | -- | --
to_ | Expensive | owned -> owned (Copy types)

Also example from guideline:

> [f64::to_radians()](https://doc.rust-lang.org/std/primitive.f64.html#method.to_radians) converts a floating point quantity from degrees to radians. The input is f64. Passing a reference &f64 is not warranted because f64 is cheap to copy. Calling the function into_radians would be misleading because the input is not consumed.

I assume `GadNodeId` to be copy type (despite generics)